### PR TITLE
feat: headers everywhere

### PR DESCRIPTION
**Pull request by pvinis**
### Description

This PR resolves [MOPLAT-408]

Using `Screen` in all the app screens!!

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-408]: https://artsyproduct.atlassian.net/browse/MOPLAT-408?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/AppProviders.tsx [hunk 1]
```diff
@@ -15,7 +15,11 @@ export const AppProviders = ({ children }: { children: React.ReactNode }) => (
         {/* @ts-expect-error */}
         <RelayEnvironmentProvider environment={defaultEnvironment}>
           <SafeAreaProvider>
-            <ProvideScreenDimensions>{children}</ProvideScreenDimensions>
+            <ProvideScreenDimensions>
+              {/*  */}
```
Comment (pvinis) [inline src/app/AppProviders.tsx:19]: for easier git diffs

File: src/app/navigation/Main.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  StorybookScreenBottomView: undefined
+  StorybookScreenFullWidthItem: undefined
+  StorybookScreenRawHeader: undefined
+}
+
+export type NavigationScreens = PreAuthScreens & RegularScreens & StorybookScreens
+
+const { Navigator, Screen } = createStackNavigator<NavigationScreens>()
+
+export const Main = () => {
```
Comment (pvinis) [inline src/app/navigation/Main.tsx:93]: everything is under react-navigation now

File: patches/@react-native-seoul+masonry-list+1.4.1.patch [hunk 3]
```diff
@@ -1,48 +1,44 @@
...
+index dc67ceb..a7a7d4b 100644
 --- a/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts
 +++ b/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts
-@@ -8,11 +8,8 @@ interface Props<T> extends Omit<ScrollViewProps, 'refreshControl'> {
+@@ -11,7 +11,7 @@ interface Props<T> extends Omit<ScrollViewProps, 'refreshControl'> {
      onEndReached?: () => void;
      onEndReachedThreshold?: number;
      style?: StyleProp<ViewStyle>;
 -    data: T[];
--    renderItem: ({ item: T, i: number }: {
```
Comment (pvinis) [inline patches/@react-native-seoul+masonry-list+1.4.1.patch:10]: they fixed some of those types, which is nice

File: src/app/App.tsx [hunk 4]
```diff
@@ -1,19 +1,8 @@
...
-
-  return (
-    <AppProviders>
-      <MainNavigationStack />
-    </AppProviders>
-  )
-}
+export const App = () => (
+  <AppProviders>
+    <Main />
```
Comment (pvinis) [inline src/app/App.tsx:6]: moving all our screens to one place

File: src/app/navigation/Main.tsx [hunk 5]
```diff
@@ -64,68 +71,97 @@ export type HomeTabsScreens = {
...
   StorybookScreenHeader: undefined
-  StorybookScreenFullWidthItem: undefined
   StorybookScreenHeaderElements: undefined
   StorybookScreenBottomView: undefined
+  StorybookScreenFullWidthItem: undefined
   StorybookScreenRawHeader: undefined
 }
 
-const { Navigator, Screen } = createStackNavigator<HomeTabsScreens>()
+export type NavigationScreens = PreAuthScreens & RegularScreens & StorybookScreens
```
Comment (pvinis) [inline src/app/navigation/Main.tsx:88]: all the screens here

File: src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/MultipleArtworksAndArtists.tsx [hunk 6]
```diff
@@ -1,34 +1,20 @@
...
-      </Screen.RawHeader>
+      <Screen.Header title="Subject lines" />
       <Screen.Body>
-        <Flex mt={1}>
-          <Text my={1} caps>
-            Multiple artworks and artists
-          </Text>
-          <Input
-            multiline
-            defaultValue={GlobalStore.useAppState(
```
Comment (pvinis) [inline src/app/screens/HomeTabs/Artists/ArtistTabs/ArtistArtworks/EmailScreen/MultipleArtworksAndArtists.tsx:26]: bad use of hook here. hooks go outside the render part.

File: src/app/sharedUI/Header.tsx [hunk 7]
```diff
@@ -1,81 +0,0 @@
...
-  rightElements?: React.ReactNode
-  safeAreaInsets?: boolean
-  positionAbsolute?: boolean
-  onPress?: () => void
-}
-
-/**
- * @deprecated Start using `Screen` with `Screen.Header`.
- */
-export const Header = ({
```
Comment (pvinis) [inline src/app/sharedUI/Header.tsx:20]: cya 👋

Comment (rajsam003) [reply src/app/sharedUI/Header.tsx:20]: 🦵kick it out

**Comment by rajsam003** [comment]
Great work Pavlos,
I just did a quick QA I saw spacing/padding issues on the below screens. Could you please fix them too?
Then we are good to go I guess

image
image
image
image
image

**Comment by pvinis** [comment]
all fixed, good catch!

**Review by rajsam003 — APPROVED** [review]
👏

**Pull request merged**